src/ufuncs.c

```diff
@@ -928,6 +929,211 @@ static void create_collection_func(char **args, npy_intp *dimensions,
 }
 static PyUFuncGenericFunction create_collection_funcs[1] = {&create_collection_func};
 
+
+
+/* Define the object -> geom functions (O_Y) */
+
+static char from_wkb_dtypes[2] = {NPY_OBJECT, NPY_OBJECT};
+static void from_wkb_func(char **args, npy_intp *dimensions,
+                          npy_intp *steps, void *data)
+{
+    void *context_handle = geos_context[0];
+    GEOSGeometry *ret_ptr;
+    PyObject *in1;
+
+    GEOSWKBReader *reader;
+    unsigned char *wkb;
+    Py_ssize_t size;
+    char is_hex;
+
+    /* Create the WKB reader */
+    reader = GEOSWKBReader_create_r(context_handle);
+    if (reader == NULL) {
+        return;
+    }
+
+    UNARY_LOOP {
+        in1 = *(PyObject **)ip1;
```

Comment:
@caspervdw can you explain this line? Why the double `**` ?

Getting this right was what I first didn't get to work, but in the end found this in one of the other functions you already wrote with object dtype input.

Reply:
Yes, I struggled with that too. The `(Pyobject **)` tells the interpreter that `ip1` is a pointer to a `PyObject *`. 

Then we take the value of that (with the extra star at the start), so that `in1` becomes a `PyObject *`.